db/migrate/20170329064703_create_user_emails.rb

```diff
@@ -0,0 +1,31 @@
+class CreateUserEmails < ActiveRecord::Migration
+  def up
+    create_table :user_emails do |t|
+      t.integer :user_id
+      t.string :email, limit: 513
+      t.timestamps
+    end
+    add_index :user_emails, :user_id
+    execute "CREATE UNIQUE INDEX index_user_emails_on_email ON user_emails ((lower(email)));"
+    execute "INSERT INTO user_emails (user_id, email, created_at, updated_at) SELECT id, email, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP FROM users WHERE users.id != -1"
+    change_table :users do |t|
+      t.remove :email
```

Comment:
Also because of this I've had to change code in a few places to explicitly specify the `user_emails.email` column rather than just `email` (which is now ambiguous)